sim/battle.ts

```diff
@@ -2228,11 +2228,9 @@ export class Battle {
 		if (this.activePerHalf > 2) {
 			if (move.target === 'adjacentFoe' || move.target === 'normal' || move.target === 'randomNormal') {
 				// even if a move can target an ally, auto-resolution will never make it target an ally
-				// i.e. if both your opponents faint before you use Flamethrower, it will fail instead of targeting your all
+				// i.e. if both your opponents faint before you use Flamethrower, it will fail instead of targeting your ally
 				const adjacentFoes = pokemon.adjacentFoes();
-				if (adjacentFoes.length) return this.sample(adjacentFoes);
-				// no valid target at all, return a possibly-fainted foe for any possible redirection
-				return pokemon.side.foe.active[0];
+				return adjacentFoes.length ? this.sample(adjacentFoes) : null;
```

Comment:
Is it safe to return `null` here? The old code (i.e. before #7929 broke it) returned the slot opposite the attacker.
https://github.com/smogon/pokemon-showdown/blob/39f5717c255cef59ca32c9351b5919032729e064/sim/battle.ts#L2056-L2062

Reply:
Given that all the tests pass I would like to say yes. This code only runs for triple battles and something like `any` will skip to the last return statement where `Side#randomFoe()` is called.

Reply:
So would the storm drain user be in range or out of range? and should the storm drain successfully retarget the move?

Reply:
Ok I did that test locally and yeah Storm Drain isn't redirecting when it should. I'm not really sure what you would return here then.

Reply:
Return the Pokémon in front of the attacker. The old code needs to be properly updated to account for more than 2 sides though.